File: packages/paysafe-card-payments/android/src/main/java/com/paysafecardpayments/expiryDatePicker/PSExpiryDatePickerViewManager.kt
// Copyright Paysafe 2025. All rights reserved.

package com.paysafecardpayments.expiryDatePicker

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext

class PSExpiryDatePickerViewManager(
  private val reactContext: ReactApplicationContext,
  private val viewFactory: (android.app.Activity) -> PSExpiryDatePickerWrapperView = { PSExpiryDatePickerWrapperView(it) }
) : SimpleViewManager<PSExpiryDatePickerWrapperView>() {

  override fun getName(): String = "PSExpiryDatePickerView"

  override fun createViewInstance(reactContext: ThemedReactContext): PSExpiryDatePickerWrapperView {
    return createWrapperView(reactContext)
  }

  internal fun createWrapperView(reactContext: ThemedReactContext): PSExpiryDatePickerWrapperView {
    val activity = reactContext.currentActivity
      ?: throw IllegalStateException("Activity is null when creating PSExpiryDatePickerWrapperView")
    return viewFactory(activity)
  }
}

File: packages/paysafe-card-payments/ios/PSExpiryDatePickerViewManager.swift
//
//  PSExpiryDatePickerViewManager.swift
//
//  Copyright © 2025 Paysafe. All rights reserved.
//

import PaysafePaymentsSDK
import React
import UIKit

@objc(PSExpiryDatePickerView) class PSExpiryDatePickerViewManager: RCTViewManager {

  override func view() -> UIView? {
    PSCardExpiryInputView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
